action.yml

```diff
@@ -0,0 +1,18 @@
+name: pysa
+description: "Run pysa on the repository and check for vulnerabilities"
+
+inputs:
+  build-type:
+    description: Pysa build to use. Possible values are stable and nightly. Default is stable.
+    required: false
+    default: "stable"
+runs:
+  using: "composite"
+  steps:
+    - run: if [ ${{inputs.build-type}} = "nighty" ]; then pip3 install pyre-check-nightly; else pip3 install pyre-check; fi
+      shell: bash
+    - run: if [ ! -f .pyre_configuration ]; then printf ".\n" | pyre init; fi
+      shell: bash
+    - run: pyre -n analyze --no-verify | python3 ${{github.action_path}}/scripts/convert_pysa_results_to_sarif.py > sarif.json
```

Comment:
Internally @0xedward is working on supporting sarif output directly inside SAPP. This allow us to use filters and not simply output all the issues. @abishekvashok we can put this diff on hold until we have this landed and then we can use the new SAPP feature to output SARIF results and then make use of it here. This will probably mean that instead of calling "python3 ${{github.action_path}}/scripts/convert_pysa_results_to_sarif.py " we'll call sapp with some parameter. What do you think?

Reply:
Makes sense to me!